mathics/eval/plot.py

```diff
@@ -120,6 +124,149 @@ def quiet_f(*args):
     return quiet_f
 
 
+def eval_ListPlot(
+    plot_points: list,
+    x_range: int,
+    y_range: int,
+    is_discrete_plot: bool,
+    is_joined_plot: bool,
+    filling,
+    options: dict,
+):
+    if isinstance(plot_points, list) and len(plot_points) != 0:
+        if all(not isinstance(point, (list, tuple)) for point in plot_points):
+            # Only y values given
+            plot_points = [
+                [[float(i + 1), plot_points[i]] for i in range(len(plot_points))]
+            ]
+        elif all(
+            isinstance(line, (list, tuple)) and len(line) == 2 for line in plot_points
+        ):
+            # Single list of (x,y) pairs
+            plot_points = [plot_points]
+        elif all(isinstance(line, list) for line in plot_points):
+            # List of lines
+            if all(
+                isinstance(point, list) and len(point) == 2
+                for line in plot_points
+                for point in line
+            ):
+                pass
+            elif all(
+                not isinstance(point, list) for line in plot_points for point in line
+            ):
+                plot_points = [
+                    [[float(i + 1), l] for i, l in enumerate(line)]
+                    for line in plot_points
+                ]
+            else:
+                return
+        else:
+            return
+    else:
+        return
```

Comment:
Maybe use early returns here?

Reply:
Good point. I did one level. Doing other levels is trickier.